spec.html

```diff
@@ -89,133 +92,133 @@ <h3>Non-terminals</h3>
                   <tr class="section">
                     <td class="fix nt">element</td>
                     <td class="fix op">::=</td>
-                    <td class="fix ex">"\x01" e_name double</td>
+                    <td class="fix ex">signed_byte(1) e_name double</td>
                     <td>64-bit binary floating point</td>
                   </tr>
                   <tr>
                     <td class="fix nt"></td>
                     <td class="fix op">|</td>
-                    <td class="fix ex">"\x02" e_name string</td>
+                    <td class="fix ex">2 e_name string</td>
                     <td>UTF-8 string</td>
                   </tr>
                   <tr>
                     <td class="fix nt"></td>
                     <td class="fix op">|</td>
-                    <td class="fix ex">"\x03"&nbsp;e_name&nbsp;document</td>
-                    <td>Embedded document</td>
+                    <td class="fix ex">signed_byte(3) e_name document</td>
+                    <td>Embedded document or object</td>
                   </tr>
                   <tr>
                     <td class="fix nt"></td>
                     <td class="fix op">|</td>
-                    <td class="fix ex">"\x04" e_name document</td>
+                    <td class="fix ex">signed_byte(4) e_name document</td>
                     <td >Array</td>
                   </tr>
                   <tr>
                     <td class="fix nt"></td>
                     <td class="fix op">|</td>
-                    <td class="fix ex">"\x05" e_name binary</td>
+                    <td class="fix ex">signed_byte(5) e_name binary</td>
                     <td>Binary data</td>
                   </tr>
                   <tr>
                     <td class="fix nt"></td>
                     <td class="fix op">|</td>
-                    <td class="fix ex">"\x06" e_name</td>
+                    <td class="fix ex">signed_byte(6) e_name</td>
                     <td>Undefined (value) &mdash; <em>Deprecated</em></td>
                   </tr>
                   <tr>
                     <td class="fix nt"></td>
                     <td class="fix op">|</td>
-                    <td class="fix ex">"\x07" e_name (byte*12)</td>
-                    <td><a href="http://dochub.mongodb.org/core/objectids">ObjectId</a></td>
+                    <td class="fix ex">signed_byte(7) e_name (byte*12)</td>
+                    <td><a href="https://www.mongodb.com/docs/manual/reference/bson-types/#objectid">ObjectId</a></td>
                   </tr>
                   <tr>
                     <td class="fix nt"></td>
                     <td class="fix op">|</td>
-                    <td class="fix ex">"\x08" e_name "\x00"</td>
-                    <td>Boolean "false"</td>
+                    <td class="fix ex">signed_byte(8) e_name 0</td>
```

Comment:
Oo. Should we say `signed_byte(0)`? I think we should. That is the bit pattern 0000 0000. Since this is a wire format and storage format, it's important to be very precise.

My only hesitation is that whether it's a signed or unsigned 0 is immaterial. Oh well, I think that's simpler than defining yet another type called `unsigned_byte`?

Similar comment for the `1` below in the definition of boolean true.

Reply:
Can replace this with `unsigned_byte(0)` and `unsigned_byte(1)` for true. The immateriality of 0's sign should not be confusing for readers

Reply:
I opened a new thread about this against the latest changes, but to be clear I think for booleans you should use `unsigned_byte(0)` and `unsigned_byte(1)` for true and false respectively.